### PR TITLE
JNG-5758 Validate nested new elements on update

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5758" title="JNG-5758" target="_blank">BugJNG-5758</a>  Validation with newly created nested element throws internal server error
  </td></table>
  <br />
 

JNG-5758 Validate nested new elements on update

### CODE REVIEW

File: judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java [hunk 1]
```diff
@@ -274,11 +287,17 @@ public List<ValidationResult> validateReference(final EReference reference, fina
 
     public Collection<ValidationResult> validateAttribute(final EAttribute attribute, final Payload instance, final Map<String, Object> validationContext) {
         final Object value = instance.get(attribute.getName());
+        final String containerLocation = (String) validationContext.getOrDefault(LOCATION_KEY, "");
```
Comment (bencegelei) [inline judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java]: unused variable?

File: judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java [hunk 2]
```diff
@@ -68,11 +68,17 @@ public enum RequiredStringValidatorOption {
     public static final String VALIDATE_MISSING_FEATURES_KEY = "validateMissingFeatures";
     public static final String IGNORE_INVALID_VALUES_KEY = "ignoreInvalidValues";
 
+    public static final String VALIDATE_ROOT_MISSING_FEATURES_KEY = "validateRootMissingFeatures";
+
+    public static final String IS_ROOT_KEY = "isRootKey";
```
Comment (bencegelei) [inline judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java]: Key postfix is for showing that it's used as a key in maps. I don't think key postfix in string value is needed in "isRootKey"

**Review by bencegelei — APPROVED** [review]

File: judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java]: isRootKey is used as a key in validationContext

File: judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java]: [2004f65](https://github.com/BlackBeltTechnology/judo-runtime-core/pull/199/commits/2004f65d5d6f26d6b6727ea7d3ab870087bdf3fa)

File: judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-validator/src/main/java/hu/blackbelt/judo/runtime/core/validator/DefaultPayloadValidator.java]: [8f52346](https://github.com/BlackBeltTechnology/judo-runtime-core/pull/199/commits/8f52346453e60d18a3545f74fc24452454c7068c)

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**